updated PR template for automation

mlodic opened the pull request.

devmrfitz commented:
@mlodic For this to work, we necessarily need to PR to have more than 1 commit. Whereas our PR template actually recommends the user to have only 1 commit

mlodic commented:
you are right, thanks for pointing this out. I cry, I would like to automate all of this.

mlodic commented:
I removed the squash things because the maintainers would do that anyway so there is no need.

Plus, I added also a description about the case of the single commit.

devmrfitz commented on .github/pull_request_template.md:
```diff
@@ -1,10 +1,9 @@
+(Please add to the PR name the issue/s that this PR would close if merged by using a [Github](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) keyword. Example: `closes #999`. If your PR is made by a single commit, please add that clause in the commit too. This is all required to automate the closure of related issues.)
```
We could just say "write `closes #123` in your first commit's message". That handles all cases
